jina/serve/networking.py

```diff
@@ -41,6 +42,52 @@ def __init__(self, summary):
         self._rr_counter = 0  # round robin counter
         self.summary = summary
 
+    async def reset_connection(self, address: str):
+        """
+        Removes and then re-adds a connection.
+        Result is the same as calling :meth:`remove_connection` and then :meth:`add_connection`, but this allows for
+        handling of race condition if multiple callers reset a connection at the same time.
+
+        :param address: Target address of this connection
+        :returns: The reset connection or None if there was no connection for the given address
+        """
+        if (
+            address in self._address_to_connection_idx
+            and self._address_to_connection_idx[address] is not None
+        ):
+            # remove connection:
+            self._rr_counter = (
```

Comment:
If this code is heavily shared with `add_` and `remove_connection` I would suggest you break those into smaller shared pieces